src/discoverAutoScalingGroup.ts

```diff
@@ -0,0 +1,35 @@
+import * as sortBy from 'lodash/sortBy';
+import {AsgDiscoveryResponse, StateMachineInput} from './utils/handlerInputs';
+import {totalRunningExecutions} from "./aws/stepFunctions";
+import {getInstancesByTag} from './aws/ec2Instances';
+
+export async function handler(event: StateMachineInput): Promise<AsgDiscoveryResponse> {
+    const runningExecutionsPromise = totalRunningExecutions(event.stepFunctionArn)
+    const runningExecutions = await runningExecutionsPromise
+
+    if (runningExecutions !== 1) {
+        const error = `Expected to find one running execution (this one!) but there were ${runningExecutions}.`;
+        return { alreadyRunning: true };
+    }
+
+    if(event.asgName) {
+        console.log(`AutoScaling group ${event.asgName} specified as input. Moving on...`);
+        return { asgName: event.asgName, alreadyRunning: false };
+    }
```

Comment:
Under what circumstances will this arise? I'm guessing this is purely for legacy setups where the CloudWatch Scheduled Event has been explicitly configured to send the ASG in the state?

Reply:
I thought it would be useful to manually rotate an asg if required. I've added a comment explaining